host/greatfet/commands/greatfet_logic.py

```diff
@@ -39,16 +42,52 @@ def main():
         sys.exit(errno.ENODEV)
 
     device.apis.logic_analyzer.start()
-
     time.sleep(1)
 
-    f = open(args.filename, 'wb')
+    fd, path = tempfile.mkstemp(None, None, os.getcwd())
+    if args.output:
+        bin_file_name = args.output
+        arc_name = args.output + '.sr'
+    else:
+        bin_file_name = path[-11:]  # temporary filename is 11 characters
```

Comment:
I don't think this is the right way to do this -- for `mkstemp` to guarantee a unique file, the full path has to be used; not just the name at the end (which I don't think is always 11 characters on all platforms).

I'd just use the output of `mkstemp` as bin_file_name directly, and then use the `arcname` field of `zip.write` to change its name to something like `logic-1` (which you can then hardcode into your metadata).